a-list.js

```diff
@@ -0,0 +1,33 @@
+function arrayToList(array){
+  let entry = null;
+  for (let i = array.length-1; i >= 0; i--){
+    entry = {value: array[i], rest: entry};
+  }
+  return entry;
+}
+
+//this is the only one I had to peek at the solution.
+function listToArray(list){
+  let array = [];
+  for (let entry = list; entry; entry = entry.rest){
+    array.push(entry.value);
```

Comment:
Do you understand this?

Reply:
The for loop starts with the entire object (`list`)  and pulls out the value and pushes it to an empty array, and in the next loop the object is the value of the key `rest` in the previous object (`entry`).